example/src/tests/hooks.ts

```diff
@@ -65,6 +65,33 @@ export function registerHooksTests() {
       expect(operation).to.equal('INSERT');
     });
 
+    it('remove update hook', async () => {
+      const hookRes: string[] = []
+      db.updateHook(({ rowId, table, operation, row = {} }) => {
+        hookRes.push(operation);
+      });
+
+      const id = chance.integer();
+      const name = chance.name();
+      const age = chance.integer();
+      const networth = chance.floating();
+      db.execute(
+        'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
+        [id, name, age, networth],
+      );
+
+      db.updateHook(null)
+
+      db.execute(
+        'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
+        [id + 1, name, age, networth],
+      );
+
+      await sleep(100)
```

Comment:
Can you try with a await new Promise(resolve => resolve()) or something similar? If not this adds 100 ms latency to the tests and we only care the callback is not trigged on the next tick

Reply:
Thus remove hook methods have no callbacks, cannot be resolved after remove the hook, if with new Promise(...) would be too hacky

```ts
	it('remove rollback hook', async () => {
      let promiseResolve: any;
      let promise = new Promise(resolve => {
        promiseResolve = resolve;
      })
      const hookRes: any[] = [];
      db.rollbackHook(() => {
        hookRes.push('rollback');
        promiseResolve?.();

        promise = new Promise(resolve => {
          promiseResolve = resolve;
        })
      });

      try {
        await db.transaction(async tx => {
          throw new Error('Blah');
        });
      } catch (e) {
        // intentionally left blank
      }
      await promise

      db.rollbackHook(null)
      promiseResolve?.();

      try {
        await db.transaction(async tx => {
          throw new Error('Blah');
        });
      } catch (e) {
        // intentionally left blank
      }
      await promise

      expect(hookRes.length).to.equal(1)
    });
```

Reply:
no, that's not what I mean. I mean awaiting an empty promise to make sure the callback queue has ticked

```ts
await Promise.resolve()
```

Reply:
How about `await sleep(0)`? 
I try with await Promise.resolve() the result was not correct, but with `await sleep(0)` is right. Maybe the callback was ticked in next loop.

```js
db.execute(
  'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
  [id, name, age, networth],
);

db.updateHook(null)

db.execute(
  'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
  [id + 1, name, age, networth],
);

// if run with `await Promise.resolve()`, the expect hookRes.length = 0, is wrong, need to be 1.
await sleep(0)

expect(hookRes.length).to.equal(1)
```

Reply:
Sounds good